src/hotspot/share/jvmci/jvmciRuntime.cpp

```diff
@@ -2020,8 +2020,9 @@ void JVMCIRuntime::compile_method(JVMCIEnv* JVMCIENV, JVMCICompiler* compiler, c
     return;
   }
 
+  assert(compile_state->task()->compile_id() <= INT_MAX, "compile id too big");
   JVMCIObject result_object = JVMCIENV->call_HotSpotJVMCIRuntime_compileMethod(receiver, jvmci_method, entry_bci,
-                                                                     (jlong) compile_state, compile_state->task()->compile_id());
+                                                                     (jlong) compile_state, (int) compile_state->task()->compile_id());
```

Comment:
Should be a cast to `jint`, right?

```suggestion
                                                                     (jlong) compile_state, (jint) compile_state->task()->compile_id());
```